app/src/ui/lib/list/section-list.tsx

```diff
@@ -0,0 +1,1664 @@
+      // Legacy in the event that we don't have ResizeObserver
+      content = (
+        <AutoSizer disableWidth={true} disableHeight={true}>
+          {({ width, height }: { width: number; height: number }) =>
+            this.renderContents(width, height)
+          }
+        </AutoSizer>
+      )
+    }
+
+    return (
+      <div
+        ref={this.onRef}
+        id={this.props.id}
+        className="list"
+        aria-labelledby={this.props.ariaLabelledBy}
+        aria-label={this.props.ariaLabel}
+      >
+        {content}
+      </div>
+    )
+  }
+
+  /**
+   * Renders the react-virtualized Grid component and optionally
+   * a fake scroll bar component if running on Windows.
+   *
+   * @param width - The width of the Grid as given by AutoSizer
+   * @param height - The height of the Grid as given by AutoSizer
+   */
+  private renderContents(width: number, height: number) {
+    if (__WIN32__) {
+      return (
+        <>
+          {this.renderGrid(width, height)}
+          {this.renderFakeScroll(height)}
+        </>
+      )
+    }
+
+    return this.renderGrid(width, height)
+  }
+
+  private getRowHeight = (section: number) => {
+    const rowHeight = this.props.rowHeight
+
+    if (typeof rowHeight === 'number') {
+      return rowHeight
+    }
+
+    return (params: Index) => {
+      const index: RowIndexPath = {
+        section: section,
+        row: params.index,
+      }
+
+      return rowHeight({ index })
+    }
+  }
+
+  private onRootGridRef = (ref: Grid | null) => {
+    this.rootGrid = ref
+  }
+
+  private getOnGridRef = (section: number) => {
+    return (ref: Grid | null) => {
+      if (ref === null) {
+        this.grids.delete(section)
+      } else {
+        this.grids.set(section, ref)
+      }
+    }
+  }
+
+  private onFakeScrollRef = (ref: HTMLDivElement | null) => {
+    this.fakeScroll = ref
+  }
+
+  private getSectionScrollOffset = (section: number) =>
+    this.props.rowCount
+      .slice(0, section)
+      .reduce((height, _x, idx) => height + this.getSectionHeight(idx), 0)
+
+  private getSectionGridRenderer =
+    (width: number, height: number) => (params: IRowRendererParams) => {
+      const section = params.rowIndex
+
+      // we select the last item from the selection array for this prop
+      const sectionHeight = this.getSectionHeight(section)
+      const offset = this.getSectionScrollOffset(section)
+
+      const relativeScrollTop = Math.max(
+        0,
+        Math.min(sectionHeight, this.state.scrollTop - offset)
+      )
+
+      const ariaLabelledBy = this.props.sectionHasHeader?.(section)
+        ? this.getRowId({ section, row: 0 })
+        : undefined
+
+      return (
+        <Grid
+          key={section}
+          id={this.props.accessibleListId}
+          role="listbox"
+          ref={this.getOnGridRef(section)}
+          autoContainerWidth={true}
+          containerRole="presentation"
+          containerProps={{ 'aria-labelledby': ariaLabelledBy }}
```

Comment:
This might not be a valid solution because the header row might not be in the DOM due to virtualization 😕 I just noticed that while recording the video of this working on Windows.

Reply:
Done!